cpp/src/query/spatial_window.cu

```diff
@@ -0,0 +1,146 @@
+};
+
+// Type-dispatch functor that creates the spatial window filter of the correct type.
+// Only floating point types are supported.
+struct spatial_window_dispatch {
+  template <typename T, std::enable_if_t<std::is_floating_point<T>::value>* = nullptr>
+  std::unique_ptr<cudf::experimental::table> operator()(double left,
+                                                        double bottom,
+                                                        double right,
+                                                        double top,
+                                                        cudf::table_view const& xy,
+                                                        cudaStream_t stream,
+                                                        rmm::mr::device_memory_resource* mr)
+  {
+    auto device_x = cudf::column_device_view::create(xy.column(0), stream);
+    auto device_y = cudf::column_device_view::create(xy.column(1), stream);
+    return cudf::experimental::detail::copy_if(
+      xy, spatial_window_filter<T>{left, bottom, right, top, *device_x, *device_y}, mr, stream);
+  }
+
+  template <typename T, std::enable_if_t<not std::is_floating_point<T>::value>* = nullptr>
+  std::unique_ptr<cudf::experimental::table> operator()(double left,
+                                                        double bottom,
+                                                        double right,
+                                                        double top,
+                                                        cudf::table_view const& xy,
+                                                        cudaStream_t stream,
+                                                        rmm::mr::device_memory_resource* mr)
+  {
+    CUSPATIAL_FAIL("Only floating-point types supported");
+  }
+};
+
+}  // namespace
+
+namespace cuspatial {
+
+namespace detail {
+
+/*
+ * Return all points (x,y) that fall within a query window (x1,y1,x2,y2)
+ * see query.hpp
+ *
+ * Detail version that takes a stream.
+ */
+std::unique_ptr<cudf::experimental::table> points_in_spatial_window(
+  double left,
+  double bottom,
+  double right,
+  double top,
+  cudf::column_view const& x,
+  cudf::column_view const& y,
+  cudaStream_t stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  CUSPATIAL_EXPECTS(x.type() == y.type(), "point type mismatch between x/y arrays");
+  CUSPATIAL_EXPECTS(x.size() == y.size(), "#of points mismatch between x/y arrays");
+
+  CUSPATIAL_EXPECTS(not(x.has_nulls() || y.has_nulls()), "NULL point data not supported");
+
+  return cudf::experimental::type_dispatcher(x.type(),
+                                             spatial_window_dispatch(),
+                                             left,
+                                             bottom,
+                                             right,
+                                             top,
+                                             cudf::table_view{{x, y}},
```

Comment:
It's an anonymous namespace, so it doesn't matter, but why create a `table_view`?

Reply:
Because `copy_if` takes a `table_view`, not two `column_view`s.  But you have a point, so I pushed down the `table_view` creation to the call to `copy_if`.